### PR TITLE
Built in Matomo data structure support

### PR DESCRIPTION
**Pull request by Auz**

### CODE REVIEW

File: packages/front-end/components/HomePage/ExperimentsGetStarted.tsx [hunk 1]
```diff
@@ -119,12 +119,12 @@ const ExperimentsGetStarted = ({
...
-              allowImport &&
-              !hasSampleExperiment &&
-              importSampleData("datasource-form")
-            }
+            // importSampleData={
+            //   !hasDataSource &&
+            //   allowImport &&
+            //   !hasSampleExperiment &&
+            //   importSampleData("datasource-form")
+            // }
```
Comment (jdorn) [inline packages/front-end/components/HomePage/ExperimentsGetStarted.tsx:127]: Do we want to add a sample data option to this new modal?

File: packages/front-end/components/Settings/DataSourceSchemaChooser.tsx [hunk 2]
```diff
@@ -1,45 +1,24 @@
...
-  },
-  {
-    value: "rudderstack",
-    label: "RudderStack",
-  },
-];
+import { useState } from "react";
+import Field from "../Forms/Field";
+import { MdKeyboardArrowDown, MdKeyboardArrowUp } from "react-icons/md";
+import { eventSchemas } from "../../services/eventSchema";
```
Comment (jdorn) [inline packages/front-end/components/Settings/DataSourceSchemaChooser.tsx:7]: Changes to this file are no longer being used, right?  If so, I would revert this file to keep the original DataSourceForm UI intact if needed.

File: packages/front-end/components/Settings/NewDataSourceForm.tsx [hunk 3]
```diff
@@ -0,0 +1,558 @@
...
+  });
+  dataSourceConnections.forEach((d) => {
+    dataSourcesMap.set(d.type, d);
+  });
+  const selectedSchema = schemasMap.get(schema) || {
+    value: "custom",
+    label: "Custom",
+  };
+  useEffect(() => {
+    track("View New Datasource Form", {
```
Comment (jdorn) [inline packages/front-end/components/Settings/NewDataSourceForm.tsx:77]: Keep all of these track event names the same as the old form (when applicable).  To differentiate them, add an event property like `newDatasourceForm: true`

File: packages/front-end/components/Settings/NewDataSourceForm.tsx [hunk 4]
```diff
@@ -0,0 +1,558 @@
...
+const NewDataSourceForm: FC<{
+  data: Partial<DataSourceInterfaceWithParams>;
+  existing: boolean;
+  source: string;
+  onCancel: () => void;
+  onSuccess: (id: string) => Promise<void>;
+}> = ({ data, onSuccess, onCancel, source, existing }) => {
+  const [dirty, setDirty] = useState(false);
+  const [step, setStep] = useState(0);
+  const [loading, setLoading] = useState(false);
```
Comment (jdorn) [inline packages/front-end/components/Settings/NewDataSourceForm.tsx:44]: You shouldn't need to manage your own loading state here.  The PagedModal is supposed to show a loading overlay automatically while `validate` is running.

File: packages/front-end/components/Settings/NewDataSourceForm.tsx (hunk 3 above)
Comment (Auz) [reply packages/front-end/components/Settings/NewDataSourceForm.tsx]: done

File: packages/front-end/components/HomePage/ExperimentsGetStarted.tsx (hunk 1 above)
Comment (Auz) [reply packages/front-end/components/HomePage/ExperimentsGetStarted.tsx]: Added

File: packages/front-end/components/Settings/DataSourceSchemaChooser.tsx (hunk 2 above)
Comment (Auz) [reply packages/front-end/components/Settings/DataSourceSchemaChooser.tsx:7]: I wanted to support the same set of event trackers in either modal.

**Pull request merged**